cmd/axiom/dataset/dataset_trim.go

```diff
@@ -0,0 +1,144 @@
+package dataset
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/spf13/cobra"
+
+	"github.com/axiomhq/cli/internal/cmdutil"
+	"github.com/axiomhq/cli/pkg/surveyext"
+)
+
+type trimOptions struct {
+	*cmdutil.Factory
+
+	// Name of the dataset to update. If not supplied as a flag, which is
+	// optional, the user will be asked for it.
+	Name string `survey:"name"`
+	// Duration of the dataset to update. If not supplied as a flag, which is
+	// optional, the user will be asked for it.
+	Duration time.Duration `survey:"duration"`
+	// Force the creation and skip the confirmation prompt.
+	Force bool
+}
+
+func newTrimCmd(f *cmdutil.Factory) *cobra.Command {
+	opts := &trimOptions{
+		Factory: f,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "trim [<dataset-name>] [(-d|--duration) <max-duration>] [-f|--force]",
+		Short: "Trim a dataset to a given length",
+
+		Args:              cmdutil.PopulateFromArgs(f, &opts.Name),
+		ValidArgsFunction: cmdutil.DatasetCompletionFunc(f),
+
+		DisableFlagsInUseLine: true,
+
+		Example: heredoc.Doc(`
+			# Interactively trim a dataset:
+			$ axiom dataset trim
+
+			# Interactively trim dataset "nginx-logs":
+			$ axiom dataset trim nginx-logs
+			
+			# Trim a dataset and provide the parameters on the command-line.
+			# This trims the dataset down to the last 12 hours:
+			$ axiom dataset trim nginx-logs --duration="12h"
+		`),
+
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			if err := completeTrim(cmd.Context(), opts); err != nil {
+				return err
+			}
+			return runTrim(cmd.Context(), opts)
+		},
+	}
+
+	cmd.Flags().DurationVarP(&opts.Duration, "duration", "d", 0, "Duration to trim the dataset to")
```

Comment:
0 seems like a dangerous default

Reply:
Resolved by forcing the flag to be set. The zero value is never used. When running interactively, the user will be asked for it.